zerocopy-derive/tests/ui/derive_transparent.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+extern crate zerocopy;
+
+fn main() {}
+
+use core::marker::PhantomData;
+use zerocopy::{AsBytes, FromBytes, Unaligned};
+
+// Test generic transparent structs
+
+#[derive(AsBytes, FromBytes, Unaligned)]
+#[repr(transparent)]
+struct TransparentStruct<T> {
+    inner: T,
+    _phantom: PhantomData<()>,
+}
+
+fn assert_is_asbytes<T: AsBytes>() {}
+fn assert_is_frombytes<T: FromBytes>() {}
+fn assert_is_unaligned<T: Unaligned>() {}
+
+const _: fn() = assert_is_asbytes::<TransparentStruct<(u8, u32)>>;
```

Comment:
Can you use a type that's guaranteed to not be `AsBytes` here? `u32` could have alignment 1 on some platforms, and the layout of tuples might be guaranteed in the future. Maybe just a `struct NotAsBytes` or `MaybeUninit<u8>` or something?